numpy/core/src/npysort/mergesort.c.src

```diff
@@ -350,80 +350,70 @@ amergesort_@suff@(@type@ *v, npy_intp *tosort, npy_intp num, PyArrayObject *arr)
 
 
 static void
-npy_mergesort0(char *pl, char *pr, char *pw, char *vp, size_t size, npy_comparator cmp)
+npy_mergesort0(char *pl, char *pr, char *pw, char *vp, npy_intp elsize,
+               PyArray_CompareFunc *cmp, PyArrayObject *arr)
```

Comment:
Perhaps we could simplify this with a structure that `GENERIC_LT` takes as an argument. Something like
```
struct generic_cmp_data {
     PyArray_CompareFunc *cmp,
     PyArrayObject *arr
};

NPY_INLINE static int
GENERIC_LT(char *a, char *b, struct generic_compare_data *dat)
{
    return dat->cmp(a, b, dat->arr) < 0;
}
```

The `generic_cmp_data` could be allocated and initialized in the calling function and passed in instead of `cmp` and `arr`.